superset/charts/commands/warm_up_cache.py

```diff
@@ -43,31 +48,51 @@ def __init__(
     def run(self) -> dict[str, Any]:
         self.validate()
         chart: Slice = self._chart_or_id  # type: ignore
+
         try:
             form_data = get_form_data(chart.id, use_slice_data=True)[0]
-            if self._dashboard_id:
-                form_data["extra_filters"] = (
-                    json.loads(self._extra_filters)
-                    if self._extra_filters
-                    else get_dashboard_extra_filters(chart.id, self._dashboard_id)
-                )
-
-            if not chart.datasource:
-                raise Exception("Chart's datasource does not exist")
-
-            obj = get_viz(
-                datasource_type=chart.datasource.type,
-                datasource_id=chart.datasource.id,
-                form_data=form_data,
-                force=True,
-            )
-
-            # pylint: disable=assigning-non-slot
-            g.form_data = form_data
-            payload = obj.get_payload()
-            delattr(g, "form_data")
-            error = payload["errors"] or None
-            status = payload["status"]
+
+            if form_data.get("viz_type") in viz_types:
+                # Legacy visualizations.
```

Comment:
The problem with this approach is that we have non-legacy charts that are also present in `viz_types` under the same key like `table` and `big_number`.

Reply:
@michael-s-molina that may explain why testing with the `table` viz type was problematic.

Should the `table`, `big_number`, etc. viz types be removed from `superset.viz`? Alternatively if they’re still required, i.e., we simultaneously support both legacy and non-legacy versions, how does Superset differentiate between the two? Is it the presence of the query context?

Reply:
> Should the table, big_number, etc. viz types be removed from superset.viz? 

Yes. Let's remove them.

Reply:
See https://github.com/apache/superset/pull/24694 which is now a precursor to this PR.